### PR TITLE
Attach function metadata properties to pulsar source/sink

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Attach producer/consumer properties to identify the function/connector that is using a particular topic

### CODE REVIEW

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
@merlimat can you rebase this PR?

**Comment by merlimat** [comment]
There were some legitimate test failures on this one.  Since it's not breaking, moved to 2.2.

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java [hunk 1]
```diff
@@ -194,4 +207,12 @@ void setupSerDe() throws ClassNotFoundException {
             }
         }
     }
+
+    public static String getFunctionNameString(FunctionDetails fd) {
+        if (fd == null) {
+            return "";
+        }
+
+        return String.format("%s/%s/%s", fd.getTenant(), fd.getNamespace(), fd.getName());
```
Comment (srkukarni) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java]: I beleive there is already a utility function to get the fqn

**Pull request closed without merging**